### PR TITLE
Run tests in Valgrind on Linux

### PR DESCRIPTION
**Pull request by punchfox**

### CODE REVIEW

**Comment by pereckerdal** [comment]
Cool stuff!

File: CMakeLists.txt [hunk 1]
```diff
@@ -100,27 +100,29 @@ set(json_all_SOURCES
...
 
-if(WIN32)
-  target_compile_options(${json_library_TARGET} PRIVATE "/MT$<$<CONFIG:Debug>:d>")
-else()
-  set_source_files_properties(${json_detail_SSE42_SOURCES} PROPERTIES COMPILE_FLAGS "-msse4.2")
+option(SPOTIFY_JSON_USE_SSE42 "Build library with SSE 4.2 support (on x86 platforms)" ON)
+if(SPOTIFY_JSON_USE_SSE42)
+  if(WIN32)
+    target_compile_definitions(${json_library_TARGET} PUBLIC SPOTIFY_JSON_USE_SSE42=1)
+    target_compile_options(${json_library_TARGET} PRIVATE "/MT$<$<CONFIG:Debug>:d>")
```
Comment (fxb) [inline CMakeLists.txt]: This should not be inside ```if(SPOTIFY_JSON_USE_SSE42)```

Comment (fxb) [inline CMakeLists.txt]: What you want is:

```
if(SPOTIFY_JSON_USE_SSE42)
  target_compile_definitions(${json_library_TARGET} PUBLIC SPOTIFY_JSON_USE_SSE42=1)
  if(NOT WIN32)
    set_source_files_properties(${json_detail_SSE42_SOURCES} PROPERTIES COMPILE_FLAGS "-msse4.2")
  endif()
endif()
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -100,27 +100,29 @@ set(json_all_SOURCES
...
+set(json_library_TARGET "spotify-json")
 add_library(${json_library_TARGET} STATIC ${json_all_HEADERS} ${json_all_SOURCES})
-
 target_include_directories(${json_library_TARGET} PUBLIC ${json_INCLUDE_DIR})
 
-if(WIN32)
-  target_compile_options(${json_library_TARGET} PRIVATE "/MT$<$<CONFIG:Debug>:d>")
-else()
-  set_source_files_properties(${json_detail_SSE42_SOURCES} PROPERTIES COMPILE_FLAGS "-msse4.2")
+option(SPOTIFY_JSON_USE_SSE42 "Build library with SSE 4.2 support (on x86 platforms)" ON)
```
Comment (fxb) [inline CMakeLists.txt]: ... and x86-64 platforms.

**Pull request merged**